src/executables/util_service.c

```diff
@@ -1836,6 +1855,42 @@ process_server (int command_type, int argc, char **argv, bool show_usage, bool c
   return status;
 }
 
+/*
+ * process_server_start_single_node - start both a page and a transaction server,
+ *                                    the page server must be started before transaction server
+ */
+static int
+process_server_start_single_node (char *server_name, int command_type)
+{
+  int status = NO_ERROR;
+  int pid = 0;
+
+  const char *args_page[] = { UTIL_CUBRID_NAME, "-t", "page", server_name, NULL };
+  status = proc_execute (UTIL_CUBRID_NAME, args_page, false, false, false, &pid);
+
+  if (status == NO_ERROR && !is_server_running (CHECK_PAGE_SERVER, server_name, pid))
+    {
+      status = ER_GENERIC_ERROR;
+    }
+  print_result (PRINT_PAGE_SERVER_NAME, status, command_type);
+
+  if (status != NO_ERROR)
+    {
+      return status;
+    }
+
+  const char *args_transaction[] = { UTIL_CUBRID_NAME, "-t", "transaction", server_name, NULL };
+  status = proc_execute (UTIL_CUBRID_NAME, args_transaction, false, false, false, &pid);
+
+  if (status == NO_ERROR && !is_server_running (CHECK_TRANSACTION_SERVER, server_name, pid))
+    {
+      status = ER_GENERIC_ERROR;
```

Comment:
If only the PS can be started, shouldn't the PS be closed at this point?
What I want to say is that the system should be easy to use and can recover itself from all scenarios.
Say, upon a call to `process_server_start_single_node`, the situation can be in any of the states:
- both servers are already running
- only PS is already running
- only TS is already running
- no server is already running
It seems to be that the function only deals with the last situation but without any upfront check.
Also some error messages should be logged in case of any failure, for easier troubleshooting (both during devel/QA as well as later by DBA's).
Another aspect is the inconsistent/incorrect switch between configs where a server might already be alive - on the same port - that was not started in the "single node" config (but with a normal config).

Reply:
The single_node configuration is not for regular usage. It may be useful for testing and for trying out the CUBRID engine. Changing from single_node to other configuration or vice-versa is not a particularly interesting case.